src/sessions.ts

```diff
@@ -650,93 +696,102 @@ export class ClientSession
     options?: TransactionOptions
   ): Promise<T> {
     const MAX_TIMEOUT = 120000;
-    const startTime = now();
-
-    let committed = false;
-    let result: any;
 
-    while (!committed) {
-      this.startTransaction(options); // may throw on error
+    this.timeoutContext =
```

Comment:
Is it at all feasible to simulate legacy timeout behavior with a timeoutContext? ex:

```typescript
const _options = options != null && 'timeoutMS' in options && typeof options.timeoutMS === 'number' ? { timeoutMS: options.timeoutMS, ...this.clientOptions } : { timeoutMS: DEFAULT_TIMEOUT };
this.timeoutContext = TimeoutContext.create(_options);
```

Reply:
> Drivers MUST NOT change existing implementations to use timeoutMS=120000 for this case.

Your suggestion does not _necessarily_ go against this, but it seems to going against the spirit of it. I think that the code now very closely matches the specs psuedo code which is a nice-to-have for comparing against.